src/api/api.ts

```diff
@@ -0,0 +1,58 @@
+import qs from 'qs';
+import MovieDetailsModel from '../models/MovieDetailsModel';
+import axios, { CancelToken } from 'axios';
+
+interface MovieBackendModel {
+    id: number;
+    title: string;
+    tagline: string;
+    vote_average: number;
+    vote_count: number;
+    release_date: string;
+    poster_path: string;
+    overview: string;
+    budget: number;
+    revenue: number;
+    genres: string[];
+    runtime: number;
+}
+
+async function GetMovieDataAsync(selectedSortBy: string, searchQuery: string, selectedGenre: string, GenreList: string[], cancelToken: CancelToken) {
```

Comment:
Please, use camelCase (starts with a small letter) for functions and variables.

You can read some JS Style Guide https://www.w3schools.com/js/js_conventions.asp